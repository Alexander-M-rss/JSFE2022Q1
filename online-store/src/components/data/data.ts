export interface IItem {
  id: number;
  name: string;
  quantity: number;
  year: number;
  manufacturer: string;
  color: string;
  cams: number;
  favorite: boolean;
}

export const data: IItem[] = [
  {
    id: 1,
    name: 'Samsung Galaxy S21',
    quantity: 2,
    year: 2019,
    manufacturer: 'Samsung',
    color: 'белый',
    cams: 3,
    favorite: false,
  },
  {
    id: 2,
    name: 'Samsung Galaxy A03',
    quantity: 5,
    year: 2000,
    manufacturer: 'Samsung',
    color: 'красный',
    cams: 2,
    favorite: false,
  },
  {
    id: 3,
    name: 'Samsung Galaxy S20',
    quantity: 3,
    year: 2020,
    manufacturer: 'Samsung',
    color: 'желтый',
    cams: 3,
    favorite: false,
  },
  {
    id: 4,
    name: 'Xiaomi Poco X4 pro',
    quantity: 1,
    year: 2022,
    manufacturer: 'Xiaomi',
    color: 'желтый',
    cams: 3,
    favorite: false,
  },
  {
    id: 5,
    name: 'Xiaomi Redmi Note 11',
    quantity: 4,
    year: 2022,
    manufacturer: 'Xiaomi',
    color: 'белый',
    cams: 3,
    favorite: true,
  },
  {
    id: 6,
    name: 'Xiaomi Redmi 9C',
    quantity: 6,
    year: 2010,
    manufacturer: 'Xiaomi',
    color: 'красный',
    cams: 3,
    favorite: false,
  },
  {
    id: 7,
    name: 'Apple iPhone 11',
    quantity: 12,
    year: 2018,
    manufacturer: 'Apple',
    color: 'белый',
    cams: 3,
    favorite: true,
  },
  {
    id: 8,
    name: 'Apple iPhone 13',
    quantity: 10,
    year: 2020,
    manufacturer: 'Apple',
    color: 'красный',
    cams: 2,
    favorite: false,
  },
  {
    id: 9,
    name: 'Apple iPhone 12',
    quantity: 2,
    year: 2019,
    manufacturer: 'Apple',
    color: 'желтый',
    cams: 2,
    favorite: false,
  },
  {
    id: 10,
    name: 'Apple iPhone SE',
    quantity: 7,
    year: 2005,
    manufacturer: 'Apple',
    color: 'белый',
    cams: 1,
    favorite: false,
  },
  {
    id: 11,
    name: 'Samsung Galaxy A52',
    quantity: 5,
    year: 2021,
    manufacturer: 'Samsung',
    color: 'белый',
    cams: 3,
    favorite: true,
  },
  {
    id: 12,
    name: 'Xiaomi Redmi Note 11s',
    quantity: 10,
    year: 2022,
    manufacturer: 'Xiaomi',
    color: 'белый',
    cams: 3,
    favorite: true,
  },
  {
    id: 13,
    name: 'Xiaomi Poco M3',
    quantity: 5,
    year: 2020,
    manufacturer: 'Xiaomi',
    color: 'желтый',
    cams: 2,
    favorite: true,
  },
  {
    id: 14,
    name: 'Apple iPhone 13',
    quantity: 3,
    year: 2020,
    manufacturer: 'Apple',
    color: 'белый',
    cams: 2,
    favorite: true,
  },
  {
    id: 15,
    name: 'Apple iPhone 11',
    quantity: 3,
    year: 2018,
    manufacturer: 'Apple',
    color: 'желтый',
    cams: 3,
    favorite: true,
  },
  {
    id: 16,
    name: 'Xiaomi MI 9T Pro',
    quantity: 10,
    year: 2019,
    manufacturer: 'Xiaomi',
    color: 'красный',
    cams: 3,
    favorite: false,
  },
  {
    id: 17,
    name: 'Apple iPhone 12 mini',
    quantity: 7,
    year: 2019,
    manufacturer: 'Apple',
    color: 'красный',
    cams: 2,
    favorite: false,
  },
  {
    id: 18,
    name: 'Apple iPhone SE 2022',
    quantity: 10,
    year: 2022,
    manufacturer: 'Apple',
    color: 'красный',
    cams: 1,
    favorite: false,
  },
  {
    id: 19,
    name: 'Apple iPhone 7',
    quantity: 1,
    year: 2014,
    manufacturer: 'Apple',
    color: 'красный',
    cams: 1,
    favorite: false,
  },
  {
    id: 20,
    name: 'Xiaomi Poco M4 Pro 5G',
    quantity: 10,
    year: 2022,
    manufacturer: 'Xiaomi',
    color: 'желтый',
    cams: 3,
    favorite: false,
  },
  {
    id: 21,
    name: 'Samsung Galaxy M52',
    quantity: 3,
    year: 2021,
    manufacturer: 'Samsung',
    color: 'красный',
    cams: 3,
    favorite: true,
  },
];

export default data;
