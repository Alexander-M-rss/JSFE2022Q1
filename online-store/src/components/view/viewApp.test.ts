import { IItem } from '../data/data';
import AppView from './viewApp';

const data: IItem[] = [
  {
    id: 1,
    name: 'Samsung Galaxy S21',
    quantity: 2,
    year: 2019,
    manufacturer: 'Samsung',
    color: 'белый',
    cams: 3,
    favorite: false,
  },
  {
    id: 2,
    name: 'Samsung Galaxy A03',
    quantity: 5,
    year: 2000,
    manufacturer: 'Samsung',
    color: 'красный',
    cams: 2,
    favorite: false,
  },
  {
    id: 4,
    name: 'Xiaomi Poco X4 pro',
    quantity: 1,
    year: 2022,
    manufacturer: 'Xiaomi',
    color: 'желтый',
    cams: 3,
    favorite: false,
  },
  {
    id: 5,
    name: 'Xiaomi Redmi Note 11',
    quantity: 4,
    year: 2022,
    manufacturer: 'Xiaomi',
    color: 'белый',
    cams: 3,
    favorite: true,
  },
  {
    id: 7,
    name: 'Apple iPhone 11',
    quantity: 12,
    year: 2018,
    manufacturer: 'Apple',
    color: 'белый',
    cams: 3,
    favorite: true,
  },
  {
    id: 8,
    name: 'Apple iPhone 13',
    quantity: 10,
    year: 2020,
    manufacturer: 'Apple',
    color: 'красный',
    cams: 2,
    favorite: false,
  },
];

const dataSortedByNameAsc: IItem[] = [
  {
    id: 7,
    name: 'Apple iPhone 11',
    quantity: 12,
    year: 2018,
    manufacturer: 'Apple',
    color: 'белый',
    cams: 3,
    favorite: true,
  },
  {
    id: 8,
    name: 'Apple iPhone 13',
    quantity: 10,
    year: 2020,
    manufacturer: 'Apple',
    color: 'красный',
    cams: 2,
    favorite: false,
  },
  {
    id: 2,
    name: 'Samsung Galaxy A03',
    quantity: 5,
    year: 2000,
    manufacturer: 'Samsung',
    color: 'красный',
    cams: 2,
    favorite: false,
  },
  {
    id: 1,
    name: 'Samsung Galaxy S21',
    quantity: 2,
    year: 2019,
    manufacturer: 'Samsung',
    color: 'белый',
    cams: 3,
    favorite: false,
  },
  {
    id: 4,
    name: 'Xiaomi Poco X4 pro',
    quantity: 1,
    year: 2022,
    manufacturer: 'Xiaomi',
    color: 'желтый',
    cams: 3,
    favorite: false,
  },
  {
    id: 5,
    name: 'Xiaomi Redmi Note 11',
    quantity: 4,
    year: 2022,
    manufacturer: 'Xiaomi',
    color: 'белый',
    cams: 3,
    favorite: true,
  },
];

const dataSortedByYearAsc: IItem[] = [
  {
    id: 2,
    name: 'Samsung Galaxy A03',
    quantity: 5,
    year: 2000,
    manufacturer: 'Samsung',
    color: 'красный',
    cams: 2,
    favorite: false,
  },
  {
    id: 7,
    name: 'Apple iPhone 11',
    quantity: 12,
    year: 2018,
    manufacturer: 'Apple',
    color: 'белый',
    cams: 3,
    favorite: true,
  },
  {
    id: 1,
    name: 'Samsung Galaxy S21',
    quantity: 2,
    year: 2019,
    manufacturer: 'Samsung',
    color: 'белый',
    cams: 3,
    favorite: false,
  },
  {
    id: 8,
    name: 'Apple iPhone 13',
    quantity: 10,
    year: 2020,
    manufacturer: 'Apple',
    color: 'красный',
    cams: 2,
    favorite: false,
  },
  {
    id: 5,
    name: 'Xiaomi Redmi Note 11',
    quantity: 4,
    year: 2022,
    manufacturer: 'Xiaomi',
    color: 'белый',
    cams: 3,
    favorite: true,
  },
  {
    id: 4,
    name: 'Xiaomi Poco X4 pro',
    quantity: 1,
    year: 2022,
    manufacturer: 'Xiaomi',
    color: 'желтый',
    cams: 3,
    favorite: false,
  },
];

const dataSortedByQtyAsc: IItem[] = [
  {
    id: 4,
    name: 'Xiaomi Poco X4 pro',
    quantity: 1,
    year: 2022,
    manufacturer: 'Xiaomi',
    color: 'желтый',
    cams: 3,
    favorite: false,
  },
  {
    id: 1,
    name: 'Samsung Galaxy S21',
    quantity: 2,
    year: 2019,
    manufacturer: 'Samsung',
    color: 'белый',
    cams: 3,
    favorite: false,
  },
  {
    id: 5,
    name: 'Xiaomi Redmi Note 11',
    quantity: 4,
    year: 2022,
    manufacturer: 'Xiaomi',
    color: 'белый',
    cams: 3,
    favorite: true,
  },
  {
    id: 2,
    name: 'Samsung Galaxy A03',
    quantity: 5,
    year: 2000,
    manufacturer: 'Samsung',
    color: 'красный',
    cams: 2,
    favorite: false,
  },
  {
    id: 8,
    name: 'Apple iPhone 13',
    quantity: 10,
    year: 2020,
    manufacturer: 'Apple',
    color: 'красный',
    cams: 2,
    favorite: false,
  },
  {
    id: 7,
    name: 'Apple iPhone 11',
    quantity: 12,
    year: 2018,
    manufacturer: 'Apple',
    color: 'белый',
    cams: 3,
    favorite: true,
  },
];

describe('AppView.renderItem', () => {
  it('Should be an instance of Function', () => {
    expect(AppView.renderItem).toBeInstanceOf(Function);
  });

  it('Should return rendered unselected item', () => {
    //
    expect(AppView.renderItem(data[0])).toMatchSnapshot();
  });

  it('Should return rendered selected item', () => {
    expect(AppView.renderItem(data[0], true)).toMatchSnapshot();
  });
});

describe('AppView.sortByNameAsc', () => {
  it('Should be an instance of Function', () => {
    expect(AppView.sortByNameAsc).toBeInstanceOf(Function);
  });

  it('Should return sorted by ascending name array of items', () => {
    expect(AppView.sortByNameAsc(data.slice())).toEqual(dataSortedByNameAsc);
  });
});

describe('AppView.sortByNameDesc', () => {
  it('Should be an instance of Function', () => {
    expect(AppView.sortByNameDesc).toBeInstanceOf(Function);
  });

  it('Should return sorted by ascending name array of items', () => {
    expect(AppView.sortByNameDesc(data.slice())).toEqual(dataSortedByNameAsc.reverse());
  });
});

describe('AppView.sortByYearAsc', () => {
  it('Should be an instance of Function', () => {
    expect(AppView.sortByYearAsc).toBeInstanceOf(Function);
  });

  it('Should return sorted by descending year array of items', () => {
    expect(AppView.sortByYearAsc(data.slice())).toEqual(dataSortedByYearAsc);
  });
});

describe('AppView.sortByYearDesc', () => {
  it('Should be an instance of Function', () => {
    expect(AppView.sortByYearDesc).toBeInstanceOf(Function);
  });

  it('Should return sorted by descending year array of items', () => {
    expect(AppView.sortByYearDesc(data.slice())).toEqual(dataSortedByYearAsc.reverse());
  });
});

describe('AppView.sortByQtyAsc', () => {
  it('Should be an instance of Function', () => {
    expect(AppView.sortByQtyAsc).toBeInstanceOf(Function);
  });

  it('Should return sorted by ascending quantity array of items', () => {
    expect(AppView.sortByQtyAsc(data.slice())).toEqual(dataSortedByQtyAsc);
  });
});

describe('AppView.sortByQtyDesc', () => {
  it('Should be an instance of Function', () => {
    expect(AppView.sortByQtyDesc).toBeInstanceOf(Function);
  });

  it('Should return sorted by ascending quantity array of items', () => {
    expect(AppView.sortByQtyDesc(data.slice())).toEqual(dataSortedByQtyAsc.reverse());
  });
});

describe('AppView.filterByName', () => {
  it('Should be an instance of Function', () => {
    expect(AppView.filterByName).toBeInstanceOf(Function);
  });

  it('Should return array of visible items Ids filtered by names containing "aom" substring', () => {
    expect(AppView.filterByName(data, 'aom')).toEqual([4, 5]);
  });
});
