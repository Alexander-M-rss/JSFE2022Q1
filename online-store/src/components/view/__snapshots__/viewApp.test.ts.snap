// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppView.renderItem Should return rendered selected item 1`] = `
"
      <div class=\\"item\\" data-item-id=\\"1\\">
        <h4 class=\\"item-title\\">Samsung Galaxy S21</h4><div class=\\"item-img-container\\">
          <img src=\\"./img/items/1.jpg\\" alt=\\"Samsung Galaxy S21\\" class=\\"item-img\\">
        </div>
        <ul class=\\"item-props\\">
          <li>Количество: 2</li>
          <li>Год выхода: 2019</li>
          <li>Производитель: Samsung</li>
          <li>Цвет: белый</li>
          <li>Количество камер: 3</li>
          <li>Популярный: нет</li>
        </ul>
        <div class=\\"marker\\" title=\\"Товар добавлен в корзину\\"></div>
      </div>"
`;

exports[`AppView.renderItem Should return rendered unselected item 1`] = `
"
      <div class=\\"item\\" data-item-id=\\"1\\">
        <h4 class=\\"item-title\\">Samsung Galaxy S21</h4><div class=\\"item-img-container\\">
          <img src=\\"./img/items/1.jpg\\" alt=\\"Samsung Galaxy S21\\" class=\\"item-img\\">
        </div>
        <ul class=\\"item-props\\">
          <li>Количество: 2</li>
          <li>Год выхода: 2019</li>
          <li>Производитель: Samsung</li>
          <li>Цвет: белый</li>
          <li>Количество камер: 3</li>
          <li>Популярный: нет</li>
        </ul>
        <div class=\\"marker hidden\\" title=\\"Товар добавлен в корзину\\"></div>
      </div>"
`;
